### DISCUSSION TITLE
**Web part {webpartname} was not transformed***

### DISCUSSION THREAD

**Discussion by Thien73** [pnp/pnpframework#170] [Q&A]
Getting this message when I try to transform a custom webpart. How can I skip and not get this message when I do a transform with a custom webpart. 
anyone getting this issue with custom webparts?

**Comment by jansenbe** [comment]
Not all web parts can be translated to modern equivalents, see https://docs.microsoft.com/en-us/sharepoint/dev/transform/modernize-userinterface-site-pages-webparts for a list of mappings. When transformation happens and there's a web part that cannot be mapped then the default behavior is to put that place holder. When you're transforming a custom web part you do have some options by configuring the web part mapping file (see https://github.com/pnp/pnpframework/blob/dev/src/lib/PnP.Framework/Modernization/webpartmapping.xml). You can either: 
- Remove the place holder for unmapped web parts by deleting line 31 (`<ClientSideText Text="***Web part {Title} was not transformed***" Order="10" />`)
- Map your custom web part to a deployed SPFX based equivalent custom web part by adding a new mapping for your web part type
